hail/src/main/scala/is/hail/expr/ir/PCode.scala

```diff
@@ -221,3 +224,19 @@ class PCanonicalBaseStructCode(val pt: PBaseStruct, val a: Code[Long]) extends P
   def store(mb: EmitMethodBuilder, r: Value[Region], dst: Code[Long]): Code[Unit] =
     pt.constructAtAddress(mb, dst, r, pt, a, deepCopy = false)
 }
+
+abstract class PBinaryCode extends PCode {
+  def loadLength(): Code[Int]
+}
+
+class PCanonicalBinaryCode(val pt: PBinary, a: Code[Long]) extends PBinaryCode {
+  def code: Code[_] = a
+
+  def loadLength(): Code[Int] = pt.loadLength(a)
```

Comment:
I think this should include `loadBytes(): Code[Array[Byte]]` and `bytesAddress(): Code[Long]`.

Note, some things were called `blahOffset` for historical reasons that are really addresses, so I've been renaming them to `blahAddress` in these Pblah changes.